packages/alea-frontend/pages/api/gpt-redirect.ts

```diff
@@ -21,11 +23,23 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       queryString ? `?${queryString}` : ''
     }`;
 
+    const isBodyMethod = method?.toUpperCase() !== 'GET';
+    const userId = await getUserId(req);
+    let finalBody = body;
+    if (isBodyMethod && INJECT_USERID_APIS.has(apiname)) {
+      finalBody = { ...body, userId };
+    }
```

Comment:
this is not needed. Just pass the `headers` as is and call the `get_user_info` function in the python backend to get the user_id